### ISSUE TITLE
Custom notification sound does not play if phone was sleeping for a few minutes

### ISSUE THREAD

**Issue by tanuva** [signalapp/Signal-iOS#5556]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
My custom notification sound is not played if a message arrives after my phone was sleeping (locked, screen off) for a few minutes.

### Steps to reproduce
- Configure a custom notification sound in Signal's notification settings (chosen from files, a plain wav file, about 1.5 seconds long, 230 KB)
- Receive a message with the phone locked and sleeping for a few minutes

**Actual result:** No notification sound is played, yet the screen lights up and shows a notification from Signal.

If I wait for the screen to go dark again and send another message, it arrives silently again.

If I unlock my phone (via Face ID, without doing anything else) and receive another message then, the notification sound is played as expected. If I send another message immediately after the screen goes dark from an unlocked state, the sound is still played. Only after a few minutes it reverts to the failing behaviour again.

**Expected result:** The notification sound is played together with a notification appearing on screen.

### Device info

**Device**: iPhone 12 mini

**iOS version**: 14.4

**Signal version:** 5.6.0.14

### Link to debug log

https://debuglogs.org/1565a548a3690e3969b7f9a087fc66738fbe454640fa38497b2ae09752a47ad4.zip

Steps taken:
1. Enabled debug log
2. Immediately locked the phone
3. Waited 2-3 minutes
4. Received a silent message
5. Unlocked the phone and submitted the log

**Comment by tanuva** [comment]
This is in fact a duplicate of #4910 which was closed by stale-bot despite us having made clear that the issue still exists. This issue persists since two years.

**Comment by tanuva** [comment]
Nope, this is not stale.

**Comment by Miguel-Signal** [comment]
Could anyone still experiencing this issue please write in to support@signal.org with your debug logs so we can look into this further?

To share a debug log from Signal iOS, tap your profile, Settings, Help, Submit Debug Log, and copy the link.

**Comment by tanuva** [comment]
Okay, so this is interesting. Believe it or not, my comment yesterday was the first time in two years where I refrained from reproducing the bug before commenting "not stale". Today I fail to reproduce the problem. Even if the phone has been sleeping for a few minutes my custom sound is played. I am baffled.

...and also intrigued. Do you have an idea what it was? *Did* you fix something? Did Apple fix something?

**Comment by tanuva** [comment]
I had a custom sound configured today and things worked nicely as far as I noticed. So I shall return to my previous state: Do you have an idea what it was? 😅
